### PR TITLE
expression: implement vectorized evaluation for `builtinNullEQIntSig`

### PR DESCRIPTION
**Pull request by AerysNan**
Signed-off-by: AerysNan <aerysnan@gmail.com>

PCP #12103

### What problem does this PR solve? 

Implement vectorized evaluation for builtinNullEQSig.

### What is changed and how it works?

```
BenchmarkVectorizedBuiltinCompareFunc/builtinNullEQIntSig-VecBuiltinFunc-4                247856              4530 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinCompareFunc/builtinNullEQIntSig-NonVecBuiltinFunc-4              33778             34065 ns/op               0 B/op          0 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by AerysNan** [comment]
BTW, I find that there are two test files, i.e. `builtin_compare_vec_generated_test` and `builtin_control_vec_generated_test.go`. The former one seems to be built by go generate but NullEQInt isn't included. So I manually add the test case to the latter test file. Is it OK with that?

File: expression/builtin_compare_vec.go [hunk 1]
```diff
@@ -379,11 +381,94 @@ func (b *builtinGTIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) e
...
+		}
+		if isNull0 != isNull1 {
+			return 0
+		}
+		return -1
+	}
+	switch {
+	case isUnsigned0 && isUnsigned1:
+		for i := 0; i < n; i++ {
+			if nulls := checkNulls(i); nulls >= 0 {
```
Comment (b41sh) [inline expression/builtin_compare_vec.go]: There's some duplicate code
I think we can refactor it like this PR (12710)[https://github.com/pingcap/tidb/pull/12710]

Comment (AerysNan) [reply expression/builtin_compare_vec.go]: @b41sh Refactored using `vecCompareInt`.

**Comment by b41sh** [comment]
> BTW, I find that there are two test files, i.e. `builtin_compare_vec_generated_test` and `builtin_control_vec_generated_test.go`. The former one seems to be built by go generate but NullEQInt isn't included. So I manually add the test case to the latter test file. Is it OK with that?

It's ok. We did not implement `builtinNullEQIntSig` in generator.

**Comment by b41sh** [comment]
LGTM

**Pull request merged**